### ISSUE TITLE
Improve scanner

### ISSUE THREAD

**Issue by jum-s** [inventaire/inventaire-client#520]
After digging into #506, [quagga](https://serratus.github.io/quaggaJS/) could have some improvements:

- reduce the green box size and activate `patchSize: "small"`, since some feedback mentioned that you really need to get very close to the book to fit the whole green box
- since quagga works with CPU and not GPU, if the browser finds out that the hardware have several cores: start another instance of quagga with a different config and prefers the one which works best

**Comment by maxlath** [comment]
Given that the current mode works well for some users and some not, I wonder if we should try to rotate between the different possible modes until we find the one that works on the device

**Comment by koalou** [comment]
> reduce the green box size and activate patchSize: "small", since some feedback mentioned that you really need to get very close to the book to fit the whole green box

Yes, and QuaggaJS’s FAQ says that [the lens shouldn’t be too close to the object](https://github.com/serratus/quaggaJS?tab=readme-ov-file#barcodes-too-small). That might require an issue in itself but my phone stops focusing when the code is near enough to fit the box :-/

But also, couldn’t it be good to switch to [Quagga2](https://github.com/ericblade/quagga2)? The original hasn’t been updated in 7 years. The fork [comes from maintainers because serratus didn’t give them access to NPM before moving on](https://github.com/serratus/quaggaJS/issues/381#issuecomment-518507936)

**Comment by maxlath** [comment]
@koalou The switch to Quagga2 was done in https://github.com/inventaire/inventaire-client/pull/506

**Comment by koalou** [comment]
Damn, I only looked at the Svelte component and the link at the top of this issue, should have [followed the import](https://github.com/inventaire/inventaire-client/blob/main/app/modules/inventory/lib/scanner/quagga.ts) to see which Quagga you actually use 🤦

Well, sorry for the noise u_u'

**Comment by maxlath** [comment]
No worries :) Thank you for caring! Feel welcome to ask for help if you want to dig deeper in the issue

**Comment by maxlath** [comment]
Scanner issue reported on device: lg g8x thinq (lm-g850um), android 12. browser: firefox (136.0). The scanner can't focus

<table height="600px">
  <tr>
    <td>In the webapp:</td>
    <td>With the device camera:</td>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/f5bb54bf-7128-4a9f-80b7-9203b933d39d" /></td>
    <td><img src="https://github.com/user-attachments/assets/46e20ac0-5b75-4622-9344-68b5a35fd0f8" /></td>
  </tr>
<table>
